js/pages/itemdetails.js

```diff
@@ -436,20 +436,87 @@ pageInit.itemdetails = function(category, assetId) {
 			})
 		}
 	}
-	
+
+	if(settings.itemdetails.showRevenue) {
+		const elem = html`
+		<div class="clearfix item-field-container">
+			<div class="text-label text-overflow field-label">Revenue</div>
+			<span class="icon-robux-16x16"> </span> <span class="field-content-revenue"></div>
+		</div>`
+
+		document.$watch(".field-content-sales", desc => {
+			desc.parentNode.after(elem)
+		})
+
+		const apply = (sales, cost) => {
+			elem.$find(".field-content-revenue").textContent = FormatNumber(Math.round(sales*cost))
+			if(settings.general.robuxToUSD) {
+					const usd = RobuxToUSD(Math.round(sales*cost))
+					elem.$find(".field-content-revenue").textContent += ` ($${usd})`
+			}
+		}
+
+		if(category === "game-pass") {
+			$.fetch(`https://api.roblox.com/marketplace/game-pass-product-info?gamePassId=${assetId}`).then(async resp => {
+				if(!resp.ok) { return }
+				const data = await resp.json()
+				apply(data.Sales, (data.PriceInRobux*0.7))
+			})
+		} else if(category === "bundles") {
+			const url = "https://catalog.roblox.com/v1/catalog/items/details"
+			
+			const request = $.fetch(url, {
+				method: "POST",
+				headers: { "Content-Type": "application/json" },
+				body: JSON.stringify({
+					items: [
+						{ id: assetId, itemType: "Bundle", key: `Bundle_${assetId}` }
+					]
+				}),
+				xsrf: true
+			})
+
+			request.then(async resp => {
+				const json = await resp.json()
+				apply(json.data[0].purchaseCount, json.data[0].price)
+			})
+		} else {
+			getProductInfo(assetId).then(data => {
+				if (data.IsForSale == false){
+					elem.$find(".field-content-revenue").textContent = "Undefined for Offsale Items"
+				}
+				else if (data.IsLimited | data.IsLimitedUnique){
+					elem.$find(".field-content-revenue").textContent = "Undefined for Limited Items"
+				}
+				else if (data.Creator.Id == 1){
+					apply(data.Sales, data.PriceInRobux)
+				}
+				else{
+					const AssetTypes = [8, 18, 19, 41, 42, 43, 44, 45, 46, 47]
+					if (AssetTypes.includes(data.AssetTypeId)){
+						apply(data.Sales, data.PriceInRobux*0.3)
+					}
+					else{
+						apply(data.Sales, data.PriceInRobux*0.7)
+					}
+				}
+			})
+		}
+	}
```

Comment:
Functionally fine but doesn't follow style rules rest of the code follows. Also uses | (bitwise or) instead of || (logical or) which is not desirable, as it only works here because the booleans get casted into integers.

Style issues:
- Invalid indentation
- Operators not spaced out
- No space before opening brace
- else statements not on the same line as closing braces
- Use of == instead of ===

```suggestion
		const apply = (sales, cost) => {
			elem.$find(".field-content-revenue").textContent = FormatNumber(Math.round(sales * cost))
			if(settings.general.robuxToUSD) {
				const usd = RobuxToUSD(Math.round(sales * cost))
				elem.$find(".field-content-revenue").textContent += ` ($${usd})`
			}
		}

		if(category === "game-pass") {
			$.fetch(`https://api.roblox.com/marketplace/game-pass-product-info?gamePassId=${assetId}`).then(async resp => {
				if(!resp.ok) { return }
				const data = await resp.json()
				apply(data.Sales, data.PriceInRobux * 0.7)
			})
		} else if(category === "bundles") {
			const url = "https://catalog.roblox.com/v1/catalog/items/details"
			
			const request = $.fetch(url, {
				method: "POST",
				headers: { "Content-Type": "application/json" },
				body: JSON.stringify({
					items: [
						{ id: assetId, itemType: "Bundle", key: `Bundle_${assetId}` }
					]
				}),
				xsrf: true
			})

			request.then(async resp => {
				const json = await resp.json()
				apply(json.data[0].purchaseCount, json.data[0].price)
			})
		} else {
			getProductInfo(assetId).then(data => {
				if (!data.IsForSale) {
					elem.$find(".field-content-revenue").textContent = "Undefined for Offsale Items"
				} else if (data.IsLimited || data.IsLimitedUnique) {
					elem.$find(".field-content-revenue").textContent = "Undefined for Limited Items"
				} else if (data.Creator.Id === 1) {
					apply(data.Sales, data.PriceInRobux)
				} else {
					const AssetTypes = [8, 18, 19, 41, 42, 43, 44, 45, 46, 47]
					if (AssetTypes.includes(data.AssetTypeId)) {
						apply(data.Sales, data.PriceInRobux * 0.3)
					} else {
						apply(data.Sales, data.PriceInRobux * 0.7)
					}
				}
			})
		}
	}
```